qutebrowser/keyinput/modeparsers.py

```diff
@@ -230,3 +230,59 @@ def __init__(self, win_id, parent=None):
         super().__init__(win_id, parent, supports_count=True,
                          supports_chains=True)
         self.read_config('caret')
+
+
+class MarkKeyParser(keyparser.BaseKeyParser):
+
+    """KeyParser for set_mark and jump_mark mode.
+
+    Attributes:
+        _mode: Either KeyMode.set_mark or KeyMode.jump_mark.
+    """
+
+    def __init__(self, win_id, mode, parent=None):
+        super().__init__(win_id, parent, supports_count=False,
+                         supports_chains=False)
+        self._mode = mode
+
+    def handle(self, e):
+        """Override handle to always match the next key and create a mark.
+
+        Args:
+            e: the KeyPressEvent from Qt.
+
+        Return:
+            True if event has been handled, False otherwise.
+        """
+        if utils.keyevent_to_string(e) is None:
+            # this is a modifier key, let it pass and keep going
+            return True
+
+        key = e.text()
+
+        if not key.isalpha() and key != "'":
```

Comment:
`str.isalpha()` is locale-dependent. I wonder, do we actually want to restrict mark names? If so, something like `if key not in string.ascii_letters + "'":` would probably be better.

But maybe we should just let the user set a `ü` or `/` or whatever mark? Is there a reason not to?